pkg/dns/service.go

```diff
@@ -409,7 +243,7 @@ func (s *Service) GetManagedZoneForHost(ctx context.Context, host string, t traf
 		return &managedZones.Items[0], subDomain, nil
 	}
 
-	if err := s.controlClient.List(ctx, &managedZones, client.InNamespace(s.defaultCtrlNS), client.MatchingFields{"spec.domainName": parentDomain}); err != nil {
+	if err := s.controlClient.List(ctx, &managedZones, client.InNamespace(t.GetNamespace()), client.MatchingFields{"spec.domainName": parentDomain}); err != nil {
```

Comment:
This is exactly the same as the check above. I'd say you can just remove it, was intended as part of this anyway https://github.com/Kuadrant/multicluster-gateway-controller/issues/202. Is there a ManagedZone still created during local-setup or is that a manual step?

Reply:
OK, so the walkthrough demo must be expecting you to create your resources in the same ns, or create a new ManagedZone?